### PR TITLE
feat: added option to exclude tags

### PR DESCRIPTION
**Pull request by wakonig**

### CODE REVIEW

File: scilog/src/app/core/remote-data.service.ts [hunk 1]
```diff
@@ -119,7 +119,11 @@ export class LogbookItemDataService extends RemoteDataService {
       config.filter.tags.forEach(tag => {
         whereFilter.push({ "tags": tag })
       })
-
+    }
+    if ((config.filter.excludeTags) && (config.filter.excludeTags.length > 0)) {
```
Comment (minottic) [inline scilog/src/app/core/remote-data.service.ts:123]: would the safety operator work here? 
`config.filter.excludeTags?.length > 0`

File: scilog/src/app/logbook/widgets/widget-preferences/widget-preferences.component.ts [hunk 2]
```diff
@@ -92,35 +93,41 @@ export class WidgetPreferencesComponent implements OnInit {
     if (this.data.filter.tags && this.data.filter.tags.length > 0) {
       this.data.filter.tags.forEach((tag: string) => {
         console.log(tag);
-        this.tag.push({name: tag});
+        this.tag.push({ name: tag });
+      })
+    }
+    if (this.data.filter.excludeTags && this.data.filter.excludeTags.length > 0) {
```
Comment (minottic) [inline scilog/src/app/logbook/widgets/widget-preferences/widget-preferences.component.ts:99]: same as above, for the safety operator

**Review by minottic — APPROVED** [review]
nice new feature!

File: scilog/src/app/core/remote-data.service.ts (hunk 1 above)
Comment (wakonig) [reply scilog/src/app/core/remote-data.service.ts]: thanks, I changed it!

File: scilog/src/app/logbook/widgets/widget-preferences/widget-preferences.component.ts (hunk 2 above)
Comment (wakonig) [reply scilog/src/app/logbook/widgets/widget-preferences/widget-preferences.component.ts]: done

File: scilog/src/app/core/remote-data.service.ts [hunk 3]
```diff
@@ -119,7 +119,11 @@ export class LogbookItemDataService extends RemoteDataService {
...
+      config.filter.excludeTags.forEach(tag => {
```
[the remark this answers is not in the document]
Comment (wakonig) [reply scilog/src/app/core/remote-data.service.ts]: done

**Comment by wakonig** [comment]
Thanks Carlo! :)

**Comment by wakonig** [comment]
If you agree with the updated changes, it would be good if we could deploy it rather soonish. The current experiment could benefit from it as we are using the python sdk to automatically send data to scilog ;)

**Comment by minottic** [comment]
in principle I do, but tests are failing and I don't see tests for the new change. 
If we are aware and fine with this (which I am if we treat this as a special case) I will do the merging and open an issue to remember to fix tests and add missing ones

**Comment by wakonig** [comment]
I'll have a look at it. Don't merge it yet please

**Comment by wakonig** [comment]
should be fine now. Thanks!

**Pull request merged**